From 550b70c3283c825285890a58806e0402b4219f52 Mon Sep 17 00:00:00 2001
From: Hal Feng <hal.feng@starfivetech.com>
Date: Fri, 12 May 2023 17:33:20 +0800
Subject: [PATCH 062/210] i2c: designware: Delete SMBus functionalities

The driver didn't implement the smbus interface,
so replace the SMBus functionalities with
I2C_FUNC_SMBUS_EMUL.

Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
---
 drivers/i2c/busses/i2c-designware-core.h | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/i2c/busses/i2c-designware-core.h b/drivers/i2c/busses/i2c-designware-core.h
index 5eb130c1d671..55419e6758e8 100644
--- a/drivers/i2c/busses/i2c-designware-core.h
+++ b/drivers/i2c/busses/i2c-designware-core.h
@@ -18,12 +18,10 @@
 #include <linux/regmap.h>
 #include <linux/types.h>
 
-#define DW_IC_DEFAULT_FUNCTIONALITY		(I2C_FUNC_I2C | \
-						 I2C_FUNC_SMBUS_BYTE | \
-						 I2C_FUNC_SMBUS_BYTE_DATA | \
-						 I2C_FUNC_SMBUS_WORD_DATA | \
-						 I2C_FUNC_SMBUS_BLOCK_DATA | \
-						 I2C_FUNC_SMBUS_I2C_BLOCK)
+#define DW_IC_DEFAULT_FUNCTIONALITY	(I2C_FUNC_I2C | (I2C_FUNC_SMBUS_EMUL \
+					& ~I2C_FUNC_SMBUS_QUICK \
+					& ~I2C_FUNC_SMBUS_PROC_CALL \
+					& ~I2C_FUNC_SMBUS_PEC))
 
 #define DW_IC_CON_MASTER			BIT(0)
 #define DW_IC_CON_SPEED_STD			(1 << 1)
-- 
2.47.0

